lib/core/analytics/analytics.js

```diff
@@ -3,34 +3,56 @@
 
   var availity = root.availity;
 
-  availity.core.constant('AV_ANALYTICS', {
-    VIRTUAL_PAGE_TRACKING: true,
-    SERVICES: {
-      PIWIK: 'avPiwikAnalytics',
-      SPLUNK: 'avSplunkAnalytics'
-    },
-    EVENTS: {
-      PAGE: '$locationChangeSuccess',
-      DEFAULT: 'click'
-    },
-    PRE_FIX: /^avAnalytics(.*)$/,
-    // should ignore these since they are part of the directives API
-    IGNORE: ['avAnalyticsOn', 'avAnalyticsIf'],
-    ENV: { // not sure if this should live here
-      PROD: {
-        DOMAIN: 'apps.availity.com',
-        URL: 'https://piwik.availity.com/piwik/'
+  availity.core.provider('avAnalyticsConfig', function(AV_ANALYTICS) {
+
+    var config = {
+      VIRTUAL_PAGE_TRACKING: true,
+      SERVICES: {
+        PIWIK: 'avPiwikAnalytics',
+        SPLUNK: 'avSplunkAnalytics'
+      },
+      EVENTS: {
+        PAGE: '$locationChangeSuccess',
+        DEFAULT: 'click'
       },
-      QA: {
-        URL: 'https://qa-piwik.availity.com/piwik/'
+      PRE_FIX: /^avAnalytics(.*)$/,
+      // should ignore these since they are part of the directives API
+      IGNORE: ['avAnalyticsOn', 'avAnalyticsIf'],
+      ENV: { // not sure if this should live here
+        PROD: {
+          DOMAIN: 'apps.availity.com',
+          URL: 'https://piwik.availity.com/piwik/'
+        },
+        QA: {
+          URL: 'https://qa-piwik.availity.com/piwik/'
+        }
       }
-    }
+    };
+
+    this.set = function(options) {
+      _.merge(config, options);
+      // TODO: remove when AV_ANALYTICS is no longer supported
+      // ensures avAnalyticsConfig and AV_ANALYTICS constant stay in sync
+      angular.copy(config, AV_ANALYTICS);
+    };
+
+    this.$get = function() {
+      /*
+      * TODO: change to angular.copy and remove when AV_ANALYTICS is no longer supported.
+      * Applies AV_ANALYTICS overrides to avAnalyticsConfig provider.
+      */
+      return _.merge({}, config, AV_ANALYTICS);
+    };
   });
 
-  availity.core.provider('avAnalytics', function(AV_ANALYTICS) {
+  // DEPRECATED: use avAnalyticsConfig
+  availity.core.constant('AV_ANALYTICS', {});
```

Comment:
just make sure the constant `AV_ANALYTICS` can never be empty during any phase of the Angular lifecycle else it will be a backward incompatible change